lib/algos/stream/MergeStream.js

```diff
@@ -32,6 +32,15 @@ class MergeStream extends stream.Readable {
         }
     }
 
+    _destroy(err, callback) {
+        for (let i = 0; i < 2; ++i) {
```

Comment:
is 2 here all the items? This could use more functional code like:
```
this._streamEof.filter(x => x).forEach(stream => stream.destory())
```

Not sure where everyone stands on functional programming or not, I like it but also know it has limitations.

Reply:
In this case I'm checking `streamEof[i]` but destroying `stream[i]`, so it's useful to have the index.

Reply:
gotcha, I haven't quite gotten multi-list/array looping in javascript. In python we have zip but nothing quite like it in JS as far as I know.